controllers/ibmpowervscluster_controller_test.go

```diff
@@ -839,6 +844,799 @@ func TestIBMPowerVSClusterReconciler_delete(t *testing.T) {
 			g.Expect(clusterScope.Client.Update(ctx, powervsImage2)).To(Not(Succeed()))
 		})
 	})
+
+	t.Run("when delete TransitGateway returns error", func(t *testing.T) {
+		g := NewWithT(t)
+		clusterScope = &scope.PowerVSClusterScope{
+			Logger: klog.Background(),
+			IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "IBMPowerVSCluster",
+					APIVersion: "infrastructure.cluster.x-k8s.io/v1beta1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:        "capi-powervs-cluster",
+					Annotations: map[string]string{"powervs.cluster.x-k8s.io/create-infra": "true"},
+					OwnerReferences: []metav1.OwnerReference{
+						{
+							APIVersion: capiv1beta1.GroupVersion.String(),
+							Kind:       "Cluster",
+							Name:       "capi-test",
+							UID:        "1",
+						}},
+				},
+				Spec: infrav1beta2.IBMPowerVSClusterSpec{
+					ServiceInstanceID: "service-instance-1",
+				},
+				Status: infrav1beta2.IBMPowerVSClusterStatus{
+					ServiceInstance: &infrav1beta2.ResourceReference{
+						ID: ptr.To("serviceInstanceID"),
+					},
+					TransitGateway: &infrav1beta2.TransitGatewayStatus{
+						ID:                ptr.To("transitgatewayID"),
+						ControllerCreated: ptr.To(true),
+						PowerVSConnection: &infrav1beta2.ResourceReference{
+							ControllerCreated: ptr.To(false),
+							ID:                ptr.To("connectionID"),
+						},
+						VPCConnection: &infrav1beta2.ResourceReference{
+							ControllerCreated: ptr.To(false),
+							ID:                ptr.To("connectionID"),
+						},
+					},
+				},
+			},
+
+			Client: fake.NewClientBuilder().WithScheme(scheme.Scheme).WithObjects().Build(),
+		}
+		tgw := &tgapiv1.TransitGateway{
+			Name:   ptr.To("transitGateway"),
+			ID:     ptr.To("transitGatewayID"),
+			Status: ptr.To(string(infrav1beta2.TransitGatewayStateAvailable))}
+		response := &core.DetailedResponse{}
```

Comment:
Since nothing in response, can we inline this? (wherever applicable for other responses as well)